magma_smaht/create_metawfr.py

```diff
@@ -9,16 +9,206 @@
 ################################################
 #   Libraries
 ################################################
+
 import json, uuid
 from dcicutils import ff_utils
+import pprint
+
+pp = pprint.PrettyPrinter(indent=2)
 
 # magma
 from magma_smaht.metawfl import MetaWorkflow
+from magma_smaht.utils import (
+    get_latest_mwf,
+    get_file_set,
+    get_library_from_file_set,
+    get_sample_from_library,
+    get_mwfr_file_input_arg,
+    get_mwfr_parameter_input_arg,
+)
+
+# MetaWorkflow names are used to get the latest version.
+# We assume that they don't change!
+MWF_NAME_ILLUMINA = "Illumina_alignment_GRCh38"
+MWF_NAME_ONT = "ONT_alignment_GRCh38"
+MWF_NAME_PACBIO = "PacBio_alignment_GRCh38"
+MWF_NAME_HIC = "Hi-C_alignment_GRCh38"
+MWF_NAME_FASTQC = "short_reads_FASTQ_quality_metrics"
+
 
 ################################################
 #   Functions
 ################################################
 
+
+def mwfr_illumina_alignment(fileset_accession, length_required, smaht_key):
+    """Creates a MetaWorflowRun item in the portal for Illumina alignemnt of submitted files within a file set"""
+
+    mwf = get_latest_mwf(MWF_NAME_ILLUMINA, smaht_key)
+    print(f"Using MetaWorkflow {mwf['accession']} ({mwf['aliases'][0]})")
+
+    file_set = get_file_set(fileset_accession, smaht_key)
+    input_arg_1 = "input_files_r1_fastq_gz"
```

Comment:
maybe we can move all these argument names we are hard coding as GLOBAL variables as well like the MWF_NAME?

Reply:
Done